### PR TITLE
Update/constrain database query for user identity

### LINKED ISSUE #402: Constrain database query for user identity
The `/user/me` endpoint currently queries the database for all fields to obtain the user record. In practice, only the `role` and `status` fields are needed in the response, so the query should specify those fields to avoid using unnecessary bandwidth.

https://github.com/HackAtUCI/irvinehacks-site/blob/7e8175001f074a0e715ab6af0a8734066691173f/apps/api/src/routers/user.py#L69-L71

For better type checking, it might also help to make the return value an explicit `IdentityResponse` rather than relying on `response_model` for data filtering.

### PR DESCRIPTION
**Pull request by IanWearsHat**
Closes #402 

- Added query projection to `mongodb_handler.retrieve_one` call in `/me` route
- Replaced FastAPI `response_model` with explicit `IdentityResponse`  
- Added test to check that `IdentityResponse` fields are returned from `/me` route

### CODE REVIEW

File: apps/api/src/routers/user.py [hunk 1]
```diff
@@ -59,19 +59,21 @@ async def logout() -> RedirectResponse:
...
 
-@router.get("/me", response_model=IdentityResponse)
+@router.get("/me")
 async def me(
     user: Annotated[Union[User, None], Depends(use_user_identity)]
-) -> dict[str, object]:
+) -> IdentityResponse:
     log.info(user)
     if not user:
         return dict()
```
Comment (waalbert) [inline apps/api/src/routers/user.py:68]: Issue: return type should be IdentityResponse

**Review by waalbert — CHANGES_REQUESTED** [review]
Awesome work Ian! One thing to fix

File: apps/api/src/routers/user.py (hunk 1 above)
Comment (taesungh) [reply apps/api/src/routers/user.py:68]: Weird, mypy should have caught this, will need to troubleshoot the checks workflow.

Comment (taesungh) [reply apps/api/src/routers/user.py:68]: Was getting filtered out to only added lines, removed filter in 6c03c32.

**Review by waalbert — APPROVED** [review]
Thanks for the work Ian!

**Pull request merged**